src/components/blockchain.mjs

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+import Block from './block';
+
+import type Transaction from './transaction';
+
+class Blockchain {
+  name: string;
+  blocks: Block[];
+  pendingTransactions: Transaction[];
+  timestamp: number;
+
+  constructor(name: string) {
+    this.name = name;
+    this.blocks = [];
+    this.pendingTransactions = [];
+    this.timestamp = Date.now();
+  }
+
+  async addBlock(block: Block) {
+    this.blocks.push(block);
+  }
+
+  async addTransaction(transaction: Transaction) {
+    this.pendingTransactions.push(transaction);
+  }
+
+  async addTransactions(transactions: Transaction[]) {
+    this.pendingTransactions.push(...transactions);
+  }
+
+  async start() {
+    if (this.blocks.length !== 0) {
+      throw new Error("Can't start a non-empty Blockchain.");
+    }
+
+    const genesis = new Block('00000000000000000000000000000000');
+    await this.addBlock(genesis);
+  }
+
+  async nextBlock() {
+    if (this.blocks.length === 0) {
+      throw new Error('No previous block in Blockchain.');
+    }
+
+    if (this.pendingTransactions.length === 0) {
+      throw new Error('No new transactions to add.');
+    }
+
+    const previousHash = await this.blocks[
+      this.blocks.length - 1
+    ].computeHash();
+    const transactions = this.pendingTransactions.splice(0, 100);
```

Comment:
Why 100?

Reply:
I did it like this for now (every Block contains at most 100 transactions) but of course we can (and should) change it to take into account not the number of transactions but the sum of their size for example (like Bitcoin with each Block no bigger than 1MB)

Reply:
Good point